### PR TITLE
Builds with Go 1.18, but tests with 1.17, 1.18 and 1.19

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This uses a test matrix to ensure the floor version of Mosn is two
behind latest Go, even if one behind latest (currently 1.18) is used to
build binaries.

See https://github.com/mosn/mosn/issues/2153

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+go 1.18
 
 require (
 	github.com/gin-gonic/gin v1.7.7
-	github.com/shirou/gopsutil v3.20.11+incompatible
-	github.com/stretchr/testify v1.7.0
-	mosn.io/pkg v0.0.0-20211217101631-d914102d1baf
+	github.com/shirou/gopsutil v3.21.11+incompatible
+	github.com/stretchr/testify v1.7.1
+	mosn.io/pkg v1.1.1-0.20221021061949-9ac98b78ab84
```
Comment (codefromthecrypt) [inline go.mod:9]: draft until next release of pkg, which shouldn't happen until https://github.com/mosn/pkg/issues/66

**Comment by codefromthecrypt** [comment]
cc @taoyuanyuan

File: example/pyroscope_rideshare/go.mod [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+	github.com/hashicorp/go-syslog v1.0.0 // indirect
+	github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475 // indirect
+	github.com/shirou/gopsutil v3.21.11+incompatible // indirect
+	github.com/tklauser/go-sysconf v0.3.10 // indirect
+	github.com/tklauser/numcpus v0.5.0 // indirect
+	github.com/yusufpapurcu/wmi v1.2.2 // indirect
+	golang.org/x/sys v0.1.0 // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
+	mosn.io/api v1.1.1-0.20221020052648-537dd3d52055 // indirect
```
Comment (codefromthecrypt) [inline example/pyroscope_rideshare/go.mod:19]: this should be aligned maybe with v1.2.3

**Comment by codefromthecrypt** [comment]
@taoyuanyuan ok!

**Comment by doujiang24** [comment]
@codefromthecrypt Sorry for the delay. The test is failed, please take a look, thanks.

**Pull request closed without merging**